### PR TITLE
add `setKindOfQuantity` api

### PR DESCRIPTION
**Pull request by christophermlawson**

### CODE REVIEW

File: core/ecschema-editing/src/Editing/Properties.ts [hunk 1]
```diff
@@ -143,6 +143,26 @@ export class Properties {
     property.setCategory(new DelayedPromiseWithProps<SchemaItemKey, PropertyCategory>(categoryKey, async () => category));
   }
 
+  /**
```
Comment (ColinKerr) [inline core/ecschema-editing/src/Editing/Properties.ts:146]: How do we want to handle the validation that we have not changed the persistence unit with this edit?  If it's a breaking schema change then it's OK to change the persistence unit ... if not then it's not OK.

Also this should be consistent up and down the stack ... changing the persistence unit for one property should change it for all the overrides.

Comment (christophermlawson) [reply core/ecschema-editing/src/Editing/Properties.ts:146]: Let me look into it and I'll let you know when I have a solution commited.

Comment (christophermlawson) [reply core/ecschema-editing/src/Editing/Properties.ts:146]: I've created the backlog item for this: https://github.com/orgs/iTwin/projects/29/views/1?pane=issue&itemId=75487150

**Comment by christophermlawson** [comment]
@Mergifyio backport release/4.8x

**Comment by christophermlawson** [comment]
https://github.com/Mergifyio backport release/4.8.x

**Pull request merged**